src/modules/metadata/reducers.js

```diff
@@ -197,6 +203,58 @@ export const projects = (
         case DELETE_PROJECT_DATASET.FINISH:
             return {...state, isDeletingDataset: false};
 
+        // CREATE_PROJECT_JSON_SCHEMA
+        case CREATE_PROJECT_JSON_SCHEMA.REQUEST:
+            return {...state, isCreatingJsonSchema: true};
+        case CREATE_PROJECT_JSON_SCHEMA.RECEIVE:
+            return {
+                ...state,
+                items: state.items.map(p => p.identifier === action.data.project
+                    ? {...p, project_schemas: [...p.project_schemas, action.data]}
+                    : p
+                ),
+                itemsByID: {
+                    ...state.itemsByID,
+                    [action.data.project]: {
+                        ...(state.itemsByID[action.data.project] || {}),
+                        project_schemas: [
+                            ...((state.itemsByID[action.data.project] || {}).project_schemas || []),
+                            action.data
+                        ]
+                    }
+                }
+            };
+        case CREATE_PROJECT_JSON_SCHEMA.FINISH:
+        case CREATE_PROJECT_JSON_SCHEMA.ERROR:
```

Comment:
finish is called even if an error occurs, so don't have this case set isDeleting to false